File: select_time_slice.py
def select(site, var, start, end):
    """
    Select data from netcdf file hosted on the Princeton Hydrometeorology thredds server

    Parameters
    -----------
    site: one of the monitoring stations in quotes ('broadmead')
    var: one of the variables from this site in quotes ('Rain_1_mm_Tot'), 
         or a list of variables(['Hc', 'Hs'])
    start: starting time for data.frame ('YYYY-MM-DD hh:mm:ss')
    end: ending time for data.frame ('YYYY-MM-DD hh:mm:ss')

    Returns
    -------
    df: pandas.DataFrame object with time index and the variable(s) as the column(s)
    """    
    import xarray as xr

    data_url = 'http://hydromet-thredds.princeton.edu:9000/thredds/dodsC/MonitoringStations/'+ site+'.nc'
    ds = xr.open_dataset(data_url)
    _ds = ds[var].sel(time=slice(start, end))
    df = _ds.to_dataframe().drop(['lat','lon','station_name'], axis=1)
    ds.close()
    
    return df

File: select_time_slice.R
select <- function(site, variable, start.time, end.time){
    ## Select data from netcdf file hosted on the Princeton Hydrometeorology thredds server

    ## Inputs:
    ## site = one of the monitoring stations in quotes ('broadmead')
    ## variable = one of the variables from this site in quotes ('Rain_1_mm_Tot')
    ## start.time = starting time for data.frame ('YYYY-MM-DD hh:mm:ss')
    ## end.time = ending time for data.frame ('YYYY-MM-DD hh:mm:ss')

    ## Outputs:
    ## data.frame with columns: t and variable
    
    library('ncdf4')
    
    data_url <- paste0('http://hydromet-thredds.princeton.edu:9000/thredds/dodsC/MonitoringStations/',site,'.nc')

    ds <- nc_open(data_url)
    time.data <- ncvar_get(ds, 'time')
    time <- as.POSIXct(time.data, origin='1970-01-01', tz='UTC')

    start <- which(time == start.time)
    count <- which(time == end.time) - start
    var <- ncvar_get(ds, variable, start, count)
    t <- time[start: (start + count - 1)]
    df <- data.frame(t, var)
    colnames(df) <- c('t',variable)

    nc_close(ds)
    return(df)
    }